oximeter/db/src/native/connection.rs

```diff
@@ -6,29 +6,80 @@
 
 //! A connection and pool for talking to the ClickHouse server.
 
+use super::io::packet::client::Encoder;
+use super::io::packet::server::Decoder;
+use super::packets::client::Packet as ClientPacket;
+use super::packets::client::Query;
+use super::packets::client::QueryResult;
+use super::packets::client::OXIMETER_HELLO;
+use super::packets::client::VERSION_MAJOR;
+use super::packets::client::VERSION_MINOR;
+use super::packets::client::VERSION_PATCH;
+use super::packets::server::Hello as ServerHello;
 use super::packets::server::Packet as ServerPacket;
-use super::packets::{
-    client::{Packet as ClientPacket, Query, QueryResult},
-    server::Progress,
-};
-use super::{
-    io::packet::{client::Encoder, server::Decoder},
-    packets::{
-        client::{OXIMETER_HELLO, VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH},
-        server::{Hello as ServerHello, REVISION},
-    },
-    Error,
-};
+use super::packets::server::Progress;
+use super::packets::server::REVISION;
+use super::Error;
 use crate::native::probes;
-use futures::{SinkExt as _, StreamExt as _};
+use futures::SinkExt as _;
+use futures::StreamExt as _;
+use qorb::backend;
+use qorb::backend::Error as QorbError;
 use std::net::SocketAddr;
-use tokio::net::{
-    tcp::{OwnedReadHalf, OwnedWriteHalf},
-    TcpStream,
-};
-use tokio_util::codec::{FramedRead, FramedWrite};
+use tokio::net::tcp::OwnedReadHalf;
+use tokio::net::tcp::OwnedWriteHalf;
+use tokio::net::TcpStream;
+use tokio_util::codec::FramedRead;
+use tokio_util::codec::FramedWrite;
 use uuid::Uuid;
 
+/// A pool of connections to a ClickHouse server over the native protocol.
+pub type Pool = qorb::pool::Pool<Connection>;
+
+/// The default connection pooling policy for native TCP connections.
+pub fn default_pool_policy() -> qorb::policy::Policy {
+    qorb::policy::Policy {
+        claim_timeout: std::time::Duration::from_secs(10),
```

Comment:
Probably worth a comment why this is provided explicitly instead of using qorb's default.